Dependencies/xr/Source/ARCore/XR.cpp

```diff
@@ -314,6 +315,25 @@ namespace xr
 
             return task;
         }
+
+        /**
+         * Compare two poses and determine if they are approximately equal.
+         **/
+        bool operator==(const Pose& lhs, const Pose& rhs)
+        {
+            return abs(lhs.Position.X - rhs.Position.X) < FLOAT_COMPARISON_THRESHOLD
+                && abs(lhs.Position.Y - rhs.Position.Y) < FLOAT_COMPARISON_THRESHOLD
+                && abs(lhs.Position.Z - rhs.Position.Z) < FLOAT_COMPARISON_THRESHOLD
+                && abs(lhs.Orientation.X - rhs.Orientation.X) < FLOAT_COMPARISON_THRESHOLD
+                && abs(lhs.Orientation.Y - rhs.Orientation.Y) < FLOAT_COMPARISON_THRESHOLD
+                && abs(lhs.Orientation.Z - rhs.Orientation.Z) < FLOAT_COMPARISON_THRESHOLD
+                && abs(lhs.Orientation.W - rhs.Orientation.W) < FLOAT_COMPARISON_THRESHOLD;
+        }
```

Comment:
It may be fine in this case, but strictly speaking I'm not sure this is mathematically rigorous. The unit quaternion is over-described to represent 3D orientation, meaning it can (at least at times) represent the same orientation using different numbers. Consequently, this operator may yield false negatives if orientations are generated in such a way that this can occur.

Reply:
I mainly want this function to ignore very small changes to avoid spamming the consumer with micro plane adjustments that occur every frame.  I'll change it to ApproxEqual instead to make it more clear for its intended purpose.

Reply:
Gotcha. In that case, if you name it something like `WasMeaningfullyUpdated` or `ShouldUpdateConsumer` (or some other name less awful than both of those), that will remove all ambiguity both on the approximate nature of it and on the intended usage.